From 4301f2fa1c0a6be36f9c0a2377d981f175e3505c Mon Sep 17 00:00:00 2001
From: Paolo Pisati <paolo.pisati@canonical.com>
Date: Tue, 1 Jun 2021 11:01:00 +0000
Subject: [PATCH] convert drm_gem_fb_prepare_fb() to GEM atomic helpers

See also upstream:

commit 820c1707177c6fe96beed7f8cc842a683afbf890
Author: Thomas Zimmermann <tzimmermann@suse.de>
Date:   Mon Feb 22 15:17:56 2021 +0100

    drm/gem: Move drm_gem_fb_prepare_fb() to GEM atomic helpers

Signed-off-by: Paolo Pisati <paolo.pisati@canonical.com>
---
 module/evdi_modeset.c | 12 ++++++++++++
 1 file changed, 12 insertions(+)

diff --git a/module/evdi_modeset.c b/module/evdi_modeset.c
index fb1752f..f3ed72a 100644
--- a/module/evdi_modeset.c
+++ b/module/evdi_modeset.c
@@ -26,7 +26,11 @@
 #include "evdi_drm_drv.h"
 #include "evdi_cursor.h"
 #include "evdi_params.h"
+#if KERNEL_VERSION(5, 13, 0) < LINUX_VERSION_CODE
 #include <drm/drm_gem_framebuffer_helper.h>
+#else
+#include <drm/drm_gem_atomic_helper.h>
+#endif
 
 static void evdi_crtc_dpms(__always_unused struct drm_crtc *crtc,
 			   __always_unused int mode)
@@ -371,12 +375,20 @@ static void evdi_cursor_atomic_update(struct drm_plane *plane,
 
 static const struct drm_plane_helper_funcs evdi_plane_helper_funcs = {
 	.atomic_update = evdi_plane_atomic_update,
+#if KERNEL_VERSION(5, 13, 0) < LINUX_VERSION_CODE
 	.prepare_fb = drm_gem_fb_prepare_fb
+#else
+	.prepare_fb = drm_gem_plane_helper_prepare_fb
+#endif
 };
 
 static const struct drm_plane_helper_funcs evdi_cursor_helper_funcs = {
 	.atomic_update = evdi_cursor_atomic_update,
+#if KERNEL_VERSION(5, 13, 0) < LINUX_VERSION_CODE
 	.prepare_fb = drm_gem_fb_prepare_fb
+#else
+	.prepare_fb = drm_gem_plane_helper_prepare_fb
+#endif
 };
 
 static const struct drm_plane_funcs evdi_plane_funcs = {
